lib/internal/modules/esm/hooks.js

```diff
@@ -252,15 +277,22 @@ class Hooks {
       this.#validatedUrls.add(url);
     }
 
+    if (!('importAttributes' in resolution) && ('importAssertions' in resolution)) {
+      emitImportAssertionWarning();
```

Comment:
now that i read farther; it seems like you've preserved it here for `assert`, but not for `with`?